app/browser/api/ledger.js

```diff
@@ -1875,6 +1960,19 @@ const initialize = (state, paymentsEnabled) => {
     }
   }
 
+  if (!ledgerState.getInfoProp(state, 'referralTimestamp')) {
+    fs = require('fs')
+    const codePath = pathName(referralCodePath)
+
+    fs.access(codePath, fs.FF_OK, (err) => {
+      if (err) {
```

Comment:
this should set a flag in session state so that the referralCodePath check doesn't run every time a user starts Brave if we already know that they don't have a referral code. i guess you could just set `referralTimestamp` to a special value like 'none'.

Reply:
we have check above `if (!updateState.getUpdateProp(state, 'downloadId')) {`

Reply:
@NejcZdovc but isn't `!updateState.getUpdateProp(state, 'downloadId')` always true for a user who didn't install from a referral? i mean that this code shouldn't keep running on every startup for those users.

Reply:
that's correct, good catch

Reply:
But how would we know that we don't need to check for referral code?

Reply:
this file check is the only indicator that we have to know if install is from referral or not

Reply:
on thing that I can do is to set referral flag to false after we fail the first time. Because this should be only checked once that should be fine. wdyt?

Reply:
@diracdeltas

Reply:
ok did implement it b93829db3d5eaf7b0803625f225b87f5ac2b4748

Reply:
Implemented logic: we only check for that file once. And if we get error when preforming `fs.access` we will not check again